controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -948,6 +968,23 @@ func (pp *portPublisher) resolveTargetPort(subset corev1.EndpointSubset) Port {
 	return undefinedEndpointPort
 }
 
+func (pp *portPublisher) updateLocalTrafficPolicy(service corev1.Service) {
+	if pp.enableEndpointSlices {
```

Comment:
I don't think the implementation of this function is correct.  For one thing, we'll need to handle the case where endpointSlices are not enabled.  And more importantly, it doesn't look like we're actually updating the local traffic policy from the given service.

I think we can implement this just by passing a boolean directly (the local traffic policy rather than the entire service) and then doing the following:
* updating `pp.address.LocalTrafficPolicy`
* sending a `listener.Add(pp.address)` to all listeners

This should trigger the filtering and diffing logic in the endpoint translator as appropriate.

Reply:
The `LocalTrafficPolicy` is updated here
https://github.com/linkerd/linkerd2/blob/97e138fe437864030692894d9d6383acc79798b1/controller/api/destination/watcher/endpoints_watcher.go#L660
When `updateEndpointSlice` is called it will read the value from pp and create the new `AddressSet` with the value from `pp.localTrafficPolicy`. 
But I think you are correct. It can be simplified according to what you suggested. I will work on it and also adding the support for `endpoints`